bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVUtils.java

```diff
@@ -0,0 +1,129 @@
+
+    private static String processMag(final byte[] magnitude) {
+        final int length = magnitude.length;
+        if (length != 0) {
+            final BigInteger bigInteger = new BigInteger(1, magnitude);
+            final StringBuilder sb = new StringBuilder();
+            sb.append("%0");
+            sb.append(length + length);
+            sb.append("x");
+            return String.format(sb.toString(), bigInteger);
+        }
+        return "";
+    }
+
+    private static final byte[] processDigestArray(final byte[] array) {
+        int n = 0;
+        int length;
+        while (true) {
+            length = array.length;
+            if (n >= length || array[n] != 0) {
+                break;
+            }
+            ++n;
+        }
+        final int n2 = length - n;
+        final byte[] array2 = new byte[n2];
+        System.arraycopy(array, n, array2, 0, n2);
+        return array2;
+    }
+
+    public static final byte[] processDigest(byte[] digest, Certificate clientCert, Certificate serverCert) {
+        final PublicKey clientPublicKey = clientCert.getPublicKey();
+        final PublicKey serverPublicKey = serverCert.getPublicKey();
+        processMag(digest);
+        if (clientPublicKey instanceof RSAPublicKey && serverPublicKey instanceof RSAPublicKey) {
+            final RSAPublicKey clientRSAPublicKey = (RSAPublicKey) clientPublicKey;
+            final RSAPublicKey serverRSAPublicKey = (RSAPublicKey) serverPublicKey;
+            try {
+                final MessageDigest instance = MessageDigest.getInstance("SHA-256");
+                final byte[] byteArray1 = clientRSAPublicKey.getModulus().abs().toByteArray();
+                final byte[] byteArray2 = clientRSAPublicKey.getPublicExponent().abs().toByteArray();
+                final byte[] byteArray3 = serverRSAPublicKey.getModulus().abs().toByteArray();
+                final byte[] byteArray4 = serverRSAPublicKey.getPublicExponent().abs().toByteArray();
+                final byte[] r1 = processDigestArray(byteArray1);
+                final byte[] r2 = processDigestArray(byteArray2);
+                final byte[] r3 = processDigestArray(byteArray3);
+                final byte[] r4 = processDigestArray(byteArray4);
+                processMag(r1);
+                processMag(r2);
+                processMag(r3);
+                processMag(r4);
+                processMag(digest);
+                instance.update(r1);
+                instance.update(r2);
+                instance.update(r3);
+                instance.update(r4);
+                instance.update(digest);
+                digest = instance.digest();
+                processMag(digest);
+            } catch (NoSuchAlgorithmException e) {
+                LOGGER.debug("NoSuchAlgorithmException Exception", e);
+            }
+        }
+        return digest;
+    }
+
+    public static String byteArrayToString(byte[] array) {
+        StringBuffer sb = new StringBuffer();
```

Comment:
There's no need of the thread-safety overehead of a `StringBuffer` here:

```suggestion
        StringBuilder sb = new StringBuilder();
```